int64 type
---
bool set_success